### PR TITLE
Reuse chown function from containers/common when U volume flag is used

### PR DESCRIPTION
**Pull request by EduardoVega**
Signed-off-by: Eduardo Vega <edvegavalerio@gmail.com>

#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

This will make the ChangeHostPathOwnership and DangerousHostPath functions exportable from containers/common, so that they can be reused by Podman and Buildah when the U flag is specified with volumes

#### Which issue(s) this PR fixes:

This is related to this Podman PR [8349](https://github.com/containers/podman/pull/8349)

#### Does this PR introduce a user-facing change?

None

```release-note

```

### CODE REVIEW

**Comment by rhatdan** [comment]
@EduardoVega Please move these functions out of Buildah into containers/common/pkg/chown.  And then we can vendor it into buildah and Podman.

We don't want to vendor in at the buildah level.

**Comment by EduardoVega** [comment]
@rhatdan I have moved the functions to containers/common/pkg/chown. I am reusing this PR I created to add the U volume flag to the parse pkg: https://github.com/containers/common/pull/420

Once merged, I will modify this PR and Podman's PR to vendor the functions

**Comment by rhatdan** [comment]
Common is merged, so this PR needs to be closed?  Or rebased?

**Comment by EduardoVega** [comment]
Needs to be rebased. I have updated the code to reuse functions from common.

**Comment by rhatdan** [comment]
LGTM
@containers/podman-maintainers PTAL

File: go.mod [hunk 1]
```diff
@@ -4,7 +4,7 @@ go 1.12
 
 require (
 	github.com/containernetworking/cni v0.8.1
-	github.com/containers/common v0.33.1
+	github.com/containers/common v0.34.0
```
Comment (TomSweeneyRedHat) [inline go.mod:7]: @rhatdan are you good with this bump?

**Comment by TomSweeneyRedHat** [comment]
LGTM as far as the changes go.  A few quick questions, probably aimed at @rhatdan .  c/common is now at 0.34.2, should be bump to that?  Should this be cherry-picked into the release-1.19 branch and finally, it says the branch is 
out of date with the base branch.  Should this be rebased (again) or should I press the shiny "Update branch" button in the GitHub web interface.

Regardless, TYVM @EduardoVega !

File: go.mod (hunk 1 above)
Comment (rhatdan) [reply go.mod:7]: We will soon go to the latest.  This is fine for now.

**Comment by rhatdan** [comment]
/lgtm
/approve

We should NOT back port this, this is a new feature and we will roll it out in the next release v1.20.* 

We want to also get this merged into podman for podman 3.1.

**Pull request merged**